sentence_transformers/sparse_encoder/SparseEncoder.py

```diff
@@ -0,0 +1,335 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Iterable
+from typing import Any, Literal
+
+import numpy as np
+import torch
+from torch import Tensor, nn
+from tqdm import trange
+
+from sentence_transformers import SentenceTransformer
+from sentence_transformers.models import Pooling, Transformer
+from sentence_transformers.similarity_functions import SimilarityFunction
+from sentence_transformers.sparse_encoder.model_card import SparseEncoderModelCardData
+from sentence_transformers.sparse_encoder.models import CSRSparsity
+from sentence_transformers.util import batch_to_device, truncate_embeddings_for_sparse
+
+logger = logging.getLogger(__name__)
+
+
+class SparseEncoder(SentenceTransformer):
+    # TODO: Check if there is no other things we need to overwrite espacially for models specificty in the init
+    # TODO: Add the proper description with associate example
+    # TODO: Clean encode implementation
+
+    # NOTE: Function available in SparseEvaluator:
+    # ---------------------------------------------Not done---------------------------------------------
+    # - start_multi_process_pool
+    # - encode_multi_process
+    # - _encode_multi_process_worker
+    # - save
+    # - save_pretrained
+    # - push_to_hub
+    # - _load_auto_model
+    # - _load_module_class_from_ref
+    # - _load_sbert_model
+    # - _no_split_modules
+    # - _keys_to_ignore_on_save
+    # - gradient_checkpointing_enable
+
+    # --------------------------------------Done-----------------------------------------------------------
+    # - __init__ (make sure nothing else need to be overrided)
+    # - encode (clean the implementation)
+    # - _update_default_model_id
+    # - load
+
+    # -----------------------------------In my opinion shouldn't be done ------------------------------------
+    # - get_backend
+    # - forward
+    # - similarity_fn_name
+    # - similarity
+    # - similarity_pairwise
+    # - stop_multi_process_pool
+    # - set_pooling_include_prompt
+    # - get_max_seq_length
+    # - tokenize
+    # - get_sentence_features
+    # - get_sentence_embedding_dimension
+    # - truncate_sentence_embeddings
+    # - _first_module
+    # - _last_module
+    # - _create_model_card
+    # - save_to_hub
+    # - _text_length
+    # - evaluate
+    # - device
+    # - tokenizer
+    # - max_seq_length
+    # - _target_device
+
+    # -----------------------------------Added------------------------------------
+    # - set_topk
+    # - get_sparsity_stats
+
```

Comment:
```suggestion
```

Reply:
Oops, the `get_sparsity_stats` is still there. And I saw you just removed `set_topk` in a commit here.